Modules/Module_Processing/Module_Structural/xASL_wrp_LinearReg_T1w2MNI.m

```diff
@@ -53,34 +52,66 @@ function xASL_wrp_LinearReg_T1w2MNI(x, bAutoACPC)
 end
-% Add ASL images to the registration list
-% for iSess = 1:x.dataset.nSessions
-%     OtherList{end+1,1} = x.P.Path_ASL4D;
-%     OtherList{end+1,1} = x.P.Path_M0;
-%     OtherList{end+1,1} = x.P.Path_ASL4D_RevPE;
-%     OtherList{end+1,1} = x.P.Path_M0_RevPE;
     
-    % Check for other ScanTypes
-    SubjectsFolders = xASL_adm_GetFileList(x.dir.SUBJECTDIR, '^.*', 'FPList', [0 Inf],true); % true for folders
-    for iOther=1:length(SubjectsFolders)
-        DirOther = SubjectsFolders{iOther};
-        if exist(DirOther,'dir')
-            Path_Other = xASL_adm_GetFileList(DirOther, '^(?!y_).*\.nii$', 'FPList', [0 Inf]); % any DWI or func NIfTI (quick & dirty)
-            for iScan=1:length(Path_Other)
-                OtherList{end+1,1} = Path_Other{iScan};
-            end
+% Check for other ScanTypes that need to be in alignment of the T1w
+% Here, we assume that all NIfTIs from all other scan types (e.g., ASL, fMRI, DTI, etc)
+% need to remain aligned with the T1w
+otherFolders = xASL_adm_GetFileList(x.dir.SUBJECTDIR, '^.*', 'FPList', [0 Inf],true); % true for folders
+for iOther=1:length(otherFolders)
+    DirOther = otherFolders{iOther};
+    if exist(DirOther, 'dir')
+        Path_Other = xASL_adm_GetFileList(DirOther, '^(?!y_).*\.nii$', 'FPList', [0 Inf]); % any DWI or func NIfTI (quick & dirty)
+        for iScan=1:length(Path_Other)
+            OtherList{end+1,1} = Path_Other{iScan};
         end
     end
-% end
-
+end
 
 
 %% ---------------------------------------------------------------------------------------------------
-%% 3)Perform the registration
+%% 2. Issue warnings if any of the orientation matrices were changed (e.g. in an earlier processing run)
+% A potential error can occur if for whatever reason, in a previous rerun some where realigned (e.g., to MNI) and others were not.
+% So we will now issue a warning if any of the orientation matrices were changed (e.g. in an earlier processing run).
+
+% Note that this can be normal for FLAIR and its derivative WMH_SEGM, as these are first resampled to T1w.nii
+% For T1w.nii, this can also be normal if the T1w was already realigned to MNI in a previous run.
+% So we check this for ASL only, by comparing the nii.mat to nii.mat0. Either all were realigned, or none. If only some were realigned, we issue a warning.
+
+% Get all ASL NIfTIs
+checkList = [];
+for iOther=1:length(OtherList)
+    if xASL_exist(OtherList{iOther})
+        [~, folderName] = fileparts(fileparts(OtherList{iOther}));
+        if ~isempty(regexp(folderName, '^ASL_\d*$'))
+            checkList{end+1} = OtherList{iOther};
+        end
+    end
+end
+
+% Check orientation differences for all ASL NIfTIs
+% between nii.mat0 (original orientation from the scanner)
+% and nii.mat (potential new orientation after registration)
+% If no registration happened, these are equal
+for iCheck=1:length(checkList)
+    nii = xASL_io_ReadNifti(checkList{iCheck});
+    matEqual(iCheck) = min(min(nii.mat==nii.mat0));
```

Comment:
This would also work. But replacing by "isequal(nii.mat, nii.mat0)" is a bit more transparent.

Reply:
Yes, fixed in e1119ede4c2447e1602b94c5d58c5c2035de455a